### PR TITLE
Ryanyae search filter 84

### PR DESCRIPTION
**Pull request by ryanyae**
## Description
added immediate filtering and keyword searching to the timeline page

## Updates
useStates and a function that handles the filtering as well as the keyword search. The filter and keyword search are two different components handled by the same function.

## Checklist

- [x] Ran `npm run lint:fix` to format code
- [x] Requested at least one reviewer
- [x] Connected PR to Trello ticket (steps below)
- [x] Addressed code review comments
- [x] Gained at least one approval for your PR

## Connecting your PR to the corresponding ticket:
- Click on the "GitHub" button found on the right of your Trello ticket
- Choose "Attach PR" from the dropdown list
- Link your GitHub account (if you haven't done so already)
- Navigate to`VCL-content-platform` and choose your PR from the dropdown.

### CODE REVIEW

**Comment by ryanyae** [comment]
[Add search bar functionality for timeline add filtering (immediate filter)](https://trello.com/c/0Gk23E8w/84-add-search-bar-functionality-for-timeline-add-filtering-immediate-filter)

**Comment by ElisaHu1** [comment]
hi ryan, how did you test it manually? I could not see the function working

File: package.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
     "yup": "^0.32.9"
   },
   "scripts": {
-    "start": "export SET NODE_OPTIONS=--openssl-legacy-provider && react-app-rewired start",
+    "start": "set NODE_OPTIONS=--openssl-legacy-provider && react-app-rewired start",
```
Comment (ElisaHu1) [inline package.json:40]: can you revert this change? i cannot compile with this new change

**Review by ElisaHu1 — DISMISSED** [review]
looks good to me, waiting for @svolterra to review

File: package.json (hunk 1 above)
Comment (ryanyae) [reply package.json:40]: Idk how these keeps happening. Is there an easier way to take this file out my commits?

Comment (ElisaHu1) [reply package.json:40]: you can put this file to gitignore. or push a change with the original command

File: src/pages/Timeline/Timeline.tsx [hunk 2]
```diff
@@ -32,49 +34,141 @@ const Timeline: React.FC<TimelineProps> = (props) => {
   //  else success = false with "success" defaulted to true
   const [success, setSuccess] = useState<boolean>(true)
 
+  // An object containing a the necessary conditions of how the user wants to filter or search for snapshots
+  // Have to make this more universal somehow, this exact same list is in TimelineFilter.tsx
+  // But this is the intial state of which snapshots to show, the list in TimeineFilter.tsx outlines which options to choose
```
[the remark this answers is not in the document]
Comment (ryanyae) [reply src/pages/Timeline/Timeline.tsx:39]: It may be easier to explain over a call, but long story short is that TimelineFilter.tsx basically has a copy of the intial state of the filterBy useState in Timeline.tsx. So if the 2 lists do not line up because it wasn't hardcoded properly than you may not see some of snapshots because it only filters for those in the filterBy. That's is why I had brought up the idea of having the elements of the list and filter use backend end stuff to make it more 'universal' across the timeline page.

File: src/components/TimelineSearchbar/TimelineSearchbar.tsx [hunk 3]
```diff
@@ -1,15 +1,33 @@
...
+
+  const textFieldRef = useRef<HTMLInputElement>(null);
+
+  const handleKeyPress = (event:React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      event.preventDefault();
+
+      setFilter({...filterBy, keyword: textFieldRef.current? textFieldRef.current.value : ""})
+    }
+  }
```
[the remark this answers is not in the document]
Comment (ryanyae) [reply src/components/TimelineSearchbar/TimelineSearchbar.tsx:28]: I can make it filter on type, but I thought that would take away from the functionality of the button on the search bar.

File: src/pages/Timeline/types.tsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+
+// Object that defines the premise of what the results should be filtered for
+//  Any of the defined items can be defined as a empty string to define that it shouldn't be filtered for that item
+export interface FilterOBJ {
```
[the remark this answers is not in the document]
Comment (ryanyae) [reply src/pages/Timeline/types.tsx:4]: Yea I agree.

**Review by ElisaHu1 — APPROVED** [review]
